src/include/uapi/ipc/pm.h

```diff
@@ -75,4 +75,49 @@ struct sof_ipc_pm_core_config {
 	uint32_t enable_mask;
 } __attribute__((packed));
 
+/* DSP subsystem power states */
+enum sof_pm_state {
+	SOF_PM_STATE_D0 = 0,	/* D0 only, also initial state */
+	SOF_PM_STATE_D3,	/* D3 DSP subsystem is power gated*/
+	/* internal FW transitions across D0ix substates incl. D0 are allowed */
+	SOF_PM_STATE_D0ix,
+};
+
+/* Flags to allow FW for applying internally clock gating policies */
+enum sof_pm_cg {
+	SOF_PM_CG_ON = 0,	/* Clock gating enabled, */
+	SOF_PM_CG_OFF,		/* Clock gating disabled */
+};
+
+/* Flags to allow FW for applying power gating policies */
+enum sof_pm_pg {
+	SOF_PM_PG_ON = 0,	/* Power gating enabled, */
+	SOF_PM_PG_OFF,		/* Power gating disabled */
+};
```

Comment:
why do we need these two definitions if we have boolean states for 'prevent_power_gating' and 'prevent_clock_gating'.

Or just like for power state should we have a bitfield that's extensible for power and clock gating?